test/end-as-callback.js

```diff
@@ -1,38 +1,32 @@
 var tap = require("tap");
 var tape = require("../");
-var trim = require('string.prototype.trim');
+var concat = require('concat-stream');
 
 tap.test("tape assert.end as callback", function (tt) {
     var test = tape.createHarness({ exit: false })
-    var tc = tap.createConsumer()
+    
+    test.createStream().pipe(concat(function (rows) {
 
-    var rows = []
-    tc.on("data", function (r) { rows.push(r) })
-    tc.on("end", function () {
-        var rs = rows.map(function (r) {
-            return r && typeof r === "object" ?
-                { id: r.id, ok: r.ok, name: trim(r.name) } :
-                r
-        })
-
-        tt.deepEqual(rs, [
-            "TAP version 13",
-            "do a task and write",
-            { id: 1, ok: true, name: "null" },
-            { id: 2, ok: true, name: "should be equal" },
-            "do a task and write fail",
-            { id: 3, ok: true, name: "null" },
-            { id: 4, ok: true, name: "should be equal" },
-            { id: 5, ok: false, name: "Error: fail" },
-            "tests 5",
-            "pass  4",
-            "fail  1"
-        ])
+        var rs = rows.toString('utf8');
 
+        tt.equal(rs,
+           'TAP version 13\n'
```

Comment:
@ljharb yeah, I saw _both_ approaches in the tests:
[`test/circular-things.js#L11-L27`](https://github.com/substack/tape/blob/092344b906cd3774ba1812d2db431e77916c9f19/test/circular-things.js#L11-L27) vs: [`test/skip.js#L11-L21`](https://github.com/substack/tape/blob/092344b906cd3774ba1812d2db431e77916c9f19/test/skip.js#L11-L21) vs: [`stackTrace.js#L29-L51`](https://github.com/substack/tape/blob/092344b906cd3774ba1812d2db431e77916c9f19/test/stackTrace.js#L29-L51) ... happy to go with _either_, which is _preferable_?